packages/components/src/utils/values.js

```diff
@@ -40,3 +40,60 @@ export function isValueEmpty( value ) {
 export function getDefinedValue( values = [], fallbackValue ) {
 	return values.find( isValueDefined ) ?? fallbackValue;
 }
+
+/**
+ * @param {string} [locale]
+ * @return {[RegExp, RegExp]} The delimiter and decimal regexp
+ */
+const getDelimiterAndDecimalRegex = ( locale ) => {
+	const formatted = Intl.NumberFormat( locale ).format( 1000.1 );
+	const delimiter = formatted[ 1 ];
+	const decimal = formatted[ formatted.length - 2 ];
+	return [
+		new RegExp( `\\${ delimiter }`, 'g' ),
+		new RegExp( `\\${ decimal }`, 'g' ),
+	];
+};
+
+// https://en.wikipedia.org/wiki/Decimal_separator#Current_standards
+const INTERNATIONAL_THOUSANDS_DELIMITER = / /g;
+
+const ARABIC_NUMERAL_LOCALES = [ 'ar', 'fa', 'ur' ];
+
+const EASTERN_ARABIC_NUMBERS = /([۰-۹]|[٠-٩])/g;
+
+/**
+ * Checks to see if a value is a numeric value (`number` or `string`).
+ *
+ * Intentionally ignores whether the thousands delimiters are only
+ * in the thousands marks.
+ *
+ * @param {any}    value
+ * @param {string} [locale]
+ * @return {boolean} Whether value is numeric.
+ */
+export function isValueNumeric( value, locale = window.navigator.language ) {
+	if ( ARABIC_NUMERAL_LOCALES.some( ( l ) => locale.startsWith( l ) ) ) {
+		locale = 'en-GB';
+		if ( EASTERN_ARABIC_NUMBERS.test( value ) ) {
+			value = value
+				.replace( /([۰-۹]|[٠-٩])/g, ( /** @type {string} */ d ) =>
+					'٠١٢٣٤٥٦٧٨٩'.indexOf( d )
```

Comment:
The numbers are slightly different for ar/ckb and fa/ps. You could see them by running `Intl.NumberFormat( 'ar' ).format( 123456789.01 );` vs `Intl.NumberFormat( 'fa' ).format( 123456789.01 );`.

Reply:
Ah, I see that now, thanks.